src/python/pants/backend/go/util_rules/sdk.py

```diff
@@ -14,14 +15,28 @@
 from pants.engine.env_vars import EnvironmentVars, EnvironmentVarsRequest
 from pants.engine.fs import EMPTY_DIGEST, CreateDigest, Digest, FileContent, MergeDigests
 from pants.engine.internals.selectors import Get, MultiGet
-from pants.engine.process import Process, ProcessResult
+from pants.engine.process import Process, ProcessCacheScope, ProcessResult
 from pants.engine.rules import collect_rules, rule
 from pants.util.frozendict import FrozenDict
 from pants.util.logging import LogLevel
 
+logger = logging.getLogger(__name__)
+
+SANDBOX_CACHE = "gopath"
+NAMED_CACHE = "gopath_cache"
+
+
+def sdk_cache() -> dict[str, str]:
+    """Returns the append_only_caches config used to populate ./gopath in a sandbox.
+
+    Other processes that build go may need to use the same cache
+    """
+    return {NAMED_CACHE: SANDBOX_CACHE}
+
 
 @dataclass(frozen=True)
 class GoSdkProcess:
+    cache_scope: ProcessCacheScope
```

Comment:
These were not in alphabetical order to begin with. Maybe move to the end of the fields since the primary attribute here remains `command`?

Reply:
Will do, though only one field - `working_dir` is breaking the alpha-order. :upside_down_face: